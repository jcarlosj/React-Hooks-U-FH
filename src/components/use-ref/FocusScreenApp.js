import React from 'react';

export const FocusScreenApp = () => {

    const handleClick = () => {
        document.querySelector( 'input' ).select();     //  document.querySelector( 'input' ).focus();
    }

    return (
        <section className="container mt-5">
            <h1>Focus Screen</h1>  
            <hr /> 

            <div className="form-group">
                <input 
                    className="form-control"
                    placeholder="Tu nombre"
                />
            </div>

            <div className="form-group">
                <button 
                    className="btn btn-outline-primary"
                    onClick={ handleClick }
                >Focus</button>
            </div>
        </section>
    )
}
