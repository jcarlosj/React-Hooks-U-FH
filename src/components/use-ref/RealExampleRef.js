import React from 'react';

export const RealExampleRef = () => {
    return (
        <section className="container mt-5">
            <h1>Real Example</h1>
            <p>Usando el Hook useRef</p>  
            <hr /> 

        </section>
    )
}
