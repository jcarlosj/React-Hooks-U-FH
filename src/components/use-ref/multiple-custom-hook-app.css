.container .container > h1 {   /* selector general de hermanos */
    font-size: 1.8rem;
}