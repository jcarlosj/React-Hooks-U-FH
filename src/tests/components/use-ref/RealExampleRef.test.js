import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import { RealExampleRef } from '../../../components/use-ref/RealExampleRef';

describe( '<RealExampleRef />', () => {

    test( 'debe desplegar correctamente', () => {

        const wrapper = shallow( <RealExampleRef /> );

        expect( wrapper ).toMatchSnapshot();

    } );
    
} );