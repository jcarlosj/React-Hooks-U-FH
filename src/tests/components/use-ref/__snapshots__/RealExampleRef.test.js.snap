// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RealExampleRef /> debe desplegar correctamente 1`] = `
<section
  className="container mt-5"
>
  <h1>
    Real Example
  </h1>
  <p>
    Usando el Hook useRef
  </p>
  <hr />
  <div
    className="form-group mt-3"
  >
    <button
      className="btn btn-secondary"
      onClick={[Function]}
    >
      Show
    </button>
  </div>
</section>
`;
